cloudcix_primitives/snapshot_hyperv.py

```diff
@@ -1,19 +1,408 @@
+
+    retval, msg_list, successful_payloads, data_dict = run_host(host, 3220, {})
+    message_list.extend(msg_list)
+
+    if not retval:
+        return retval, data_dict, message_list
+    else:
+        return True, data_dict, [f'1200: {messages[1200]}']
+
+
+def scrub(
+        domain: str,
+        host: str,
+        snapshot: str,
+        remove_subtree: bool,
+) -> Tuple[bool, str]:
+    """
+    description: Removes the VM
+
+    parameters:
+        domain:
+            description: Unique identification name for the HyperV VM on the HyperV Host.
+            type: string
+            required: true
+        host:
+            description: The dns or ipadddress of the Host on which the domain is built
+            type: string
+            required: true
+        snapshot:
+            description: Identification of the HyperV VM's snapshot on the Host.
+            type: string
+            required: true
+        remove_subtree:
+            description: Whether to delete all previous snapshots for the same domain or not
+            type: boolean
+            required: true
+
+    return:
+        description: |
+            A tuple with a boolean flag stating the build was successful or not and
+            the output or error message.
+        type: tuple
+    """
+
+    # Define message
+    messages = {
+        1100: f'Successfully scrubbed domain {domain} on host {host}',
+        3121: f'Failed to connect to the host {host} for payload read_snapshot_info',
+        3122: f'Failed to connect to the host {host} for payload remove_snapshot',
+        3123: f'Failed to remove snapshot {snapshot} of the domain {domain} from host {host}',
+        3124: f'Failed to connect to the host {host} for payload remove_subtree',
+        3125: f'Failed to remove subtree of snapshot {snapshot} of the domain {domain} from host {host}',
+    }
+
+    def run_host(host, prefix, successful_payloads):
+        rcc = SSHCommsWrapper(comms_ssh, host, 'robot')
+        fmt = HostErrorFormatter(
+            host,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads
+        )
+
+        payloads = {
+            'read_snapshot_info': f'Get-VMSnapshot -VMName {domain} -Name {snapshot} -ea SilentlyContinue',
+            'remove_snapshot':    f'Remove-VMSnapshot -VMName {domain} -Name {snapshot} ',
+            'remove_subtree':     f'Remove-VMSnapshot -VMName {domain} -Name {snapshot} -IncludeAllChildSnapshots',
+
+        }
+
+        ret = rcc.run(payloads['read_snapshot_info'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f'{prefix + 1}: {messages[prefix + 1]}'), fmt.successful_payloads
+        scrub_snapshot = True
+        if ret["payload_code"] != SUCCESS_CODE:
+            scrub_snapshot = False
+        fmt.add_successful('read_snapshot_info', ret)
+
+        if scrub_snapshot is True:
+            if remove_subtree is True:
```

Comment:
```suggestion
            if remove_subtree is False:
```

The payload in this if statement is for `remove_snapshot` and the else block is for `remove_subtree`